fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1602,8 +1602,8 @@ private AggregateExprInfo buildAggregateTuple(
                 intermediateSlotDesc.setType(intermediateType);
                 intermediateSlotDesc.setIsNullable(aggrFn.isNullable());
                 intermediateSlotDesc.setIsMaterialized(true);
-                SlotRef intermediateSlotRef = new SlotRef(aggregation.getKey().toString(), intermediateSlotDesc);
-                intermediateAggrExprs.add(intermediateSlotRef);
+                intermediateAggrExprs.add(ScalarOperatorToExpr.buildExecExpression(aggregation.getKey(),
+                        new ScalarOperatorToExpr.FormatterContext(context.getColRefToExpr())));
```

Comment:
Is the above lines about `intermediateSlotDesc` is unnecessary? 
 line 1605 
```
ScalarOperatorToExpr.buildExecExpression(aggregation.getKey(),
new ScalarOperatorToExpr.FormatterContext(context.getColRefToExpr()))
```
 seems just retrun the SlotRef created in line 1595 
```
context.getColRefToExpr().put(aggregation.getKey(), new SlotRef(aggregation.getKey().toString(), slotDesc));
```
. Is it OK?

Reply:
yes, it will return the SlotRef added in 1595, but it will go through the hack process inside the ScalarOperatorToExpr

Reply:
「Is the above lines about intermediateSlotDesc is unnecessary?
line 1605」
Seems unnecessary, I'll remove it and see what happens.

Reply:
What happens if `intermediateType` is not equal with `aggregation.getValue().getType()`?

Reply:
intermediateType has been removed now.

Reply:
It's query cache related logic, I'll revert the change.